src/main/java/com/iexec/resultproxy/ipfs/IpfsService.java

```diff
@@ -21,17 +41,24 @@ public class IpfsService implements SmartLifecycle {
     private IPFS ipfs;
     private final String multiAddress;
 
-    public IpfsService(IpfsConfig ipfsConfig) {
-        String ipfsHost = ipfsConfig.getHost();
-        String ipfsNodeIp = NetworkUtils.isIPAddress(ipfsHost) ? ipfsHost : NetworkUtils.convertHostToIp(ipfsHost);
-        this.multiAddress = "/ip4/" + ipfsNodeIp + "/tcp/" + ipfsConfig.getPort();
+    public IpfsService(final IpfsConfig ipfsConfig) {
+        try {
+            final URL ipfsUrl = new URL(ipfsConfig.getUrl());
+            final String ipfsHost = ipfsUrl.getHost();
+            int port = ipfsUrl.getPort() != -1 ? ipfsUrl.getPort() : 5001;
+            final String ipfsNodeIp = InetAddresses.isInetAddress(ipfsHost) ? ipfsHost : convertHostToIp(ipfsHost);
```

Comment:
`convertToHostIp` validate proper v4 IP addresses. It seems possible to simplify this as follows:
```suggestion
            final String ipfsNodeIp = convertHostToIp(ipfsHost);
```
This means the google guava dependency may be optional